src/main/default/lwc/voteTracker/voteTracker.js

```diff
@@ -0,0 +1,166 @@
+ */
+export default class VoteTracker extends LightningElement {
+  labels = {
+    voteFinalText,
+    voteLabel,
+    voteQuestion,
+    abandonRequestText
+  };
+  /**
+   * The label to be shown to the user.
+   * @api
+   * @type {string}
+   */
+  @api label = this.labels.voteLabel;
+
+  /**
+   * The question to be shown to the user.
+   * @api
+   * @type {string}
+   */
+  @api question = this.labels.voteQuestion;
+
+  /**
+   * The size of the component.
+   * @api
+   * @type {'small'|'big'}
+   */
+  @api size = 'small';
+
+  /**
+   * The text to be shown after the vote.
+   * @api
+   * @type {string}
+   */
+  @api finalText = this.labels.voteFinalText;
+
+  /**
+   * The label to be shown in the button to abandon a request.
+   * @api
+   * @type {string}
+   */
+  @api abandonLabel = this.labels.abandonRequestText;
+
+  /**
+   * The timeout to wait before showing the final state after voting.
+   * @type {number}
+   */
+  timeout = 2000;
+
+  /**
+   * Tells if the component is on the final state that appears after voting.
+   * @type {boolean}
+   */
+  _finalState = false;
+
+  /**
+   * The state of the positive vote button.
+   * @type {'initial'|'selected'|'neutral'}
+   */
+  _positiveState = 'initial';
+
+  /**
+   * The state of the negative vote button.
+   * @type {'initial'|'selected'|'neutral'}
+   */
+  _negativeState = 'initial';
+
+  /**
+   * Returns the css class to be given to the question.
+   * @returns {string}
+   */
+  get questionClass() {
+    return this.size === 'small'
+      ? 'slds-text-title_bold text_light'
+      : 'slds-text-heading_small';
+  }
+
+  /**
+   * Tells if the abandon request button should be shown.c/exampleCaseAssist
+   * @returns {boolean}
+   */
+  get showAbandonRequest() {
+    return this.size === 'big' && this._positiveState === 'selected';
+  }
+
+  /**
+   * Tells if the component is in the final state.
+   * @returns {boolean}
+   */
+  @api
+  get finalState() {
+    return this._finalState;
+  }
+
+  /**
+   * Tells if the the size of the component is big.
+   * @returns {boolean}
+   */
+  get isBig() {
+    return this.size === 'big';
+  }
+
+  /**
+   * Returns the state of the positive vote button.
+   * @returns {'initial'|'selected'|'neutral'}
+   */
+  get positiveState() {
+    return this._positiveState;
+  }
+
+  /**
+   * Returns the state of the negative vote button.
+   * @returns {'initial'|'selected'|'neutral'}
+   */
+  get negativeState() {
+    return this._negativeState;
+  }
+
+  /**
+   * Hamdles the click on the positive vote button.
+   * @returns {void}
+   */
+  positiveVote() {
+    if (this._positiveState === 'initial') {
+      this._positiveState = 'selected';
+      this._negativeState = 'neutral';
+      // eslint-disable-next-line @lwc/lwc/no-async-operation
+      setTimeout(() => {
+        this._finalState = true;
+      }, this.timeout);
```

Comment:
Huh? Is that a placeholder for an analytics event/server-side call?

Reply:
Actually just in the mockup of that component, The UX team précised that after clicking on a vote button, we wait few seconds before hiding the buttons and displaying a Thank you message. so this setTimeout is made to wait for few seconds before switiching the component to his final state

Reply:
What happens if I click on the different voting option during that time?
Actually now that I think about it, maybe it would be best to display the message thanking the user and NOT replace the voting icons? what if I want to change my mind and no longer can click on the icon?
Can you confirm with Audrey?

Reply:
For now when you click on a different  voting option during that time I clear the timeout andstart a new one, You can see this in latest version  of my code and not in the piece of code which is linked to this comment.
So basically now you have like 2 seconds to change your mind x) if not you can't change it forever. I will check with Audrey to confirm how we want to address  this 👌